pkg/cloud/instance.go

```diff
@@ -159,6 +167,22 @@ func (c *client) GetOrCreateVMInstance(
 
 	if len(csMachine.Spec.AffinityGroupIds) > 0 {
 		p.SetAffinitygroupids(csMachine.Spec.AffinityGroupIds)
+	} else if !(strings.ToLower(csMachine.Spec.Affinity) == "no" || csMachine.Spec.Affinity == "") {
+		ownerRef := csCtrlrUtils.GetManagementOwnerRef(machine)
+		if ownerRef == nil {
+			return errors.Errorf("Could not find management owner reference for %s/%s",
+				csMachine.Namespace, csMachine.Name)
+		}
+		name := fmt.Sprintf("Affinity-%s", ownerRef.UID)
```

Comment:
Following from today's demo, can prefix the name with something descriptive, to make the AG's role clearer for the admin or troubleshooter?

Reply:
I just added the type of affinity group `pro` or `anti` along with the cluster name.